pandas/io/common.py

```diff
@@ -324,9 +325,51 @@ def _get_handle(path, mode, encoding=None, compression=None):
         else:
             f = open(path, mode)
 
+    if memory_map and hasattr(f, 'fileno'):
+        try:
+            f = MMapWrapper(f)
+        except:  # fallback: leave file handler as is
```

Comment:
OK, I guess you're right. Still, let's only catch `Exception` and make a note about why we do this.

I'm not sure that we need to consider the current "catch anything" behavior as API but I guess it's better safe than sorry.